tests/src/ut_test_kvp.c

```diff
@@ -199,37 +226,49 @@ void test_ut_kvp_open_malloced( void )
     ut_kvp_close(pInstance);
 
     /* Negative Read Test, NULL PARAM */
-    UT_LOG_STEP("ut_kvp_open( pInstance, NULL ) - Negative");
-    status = ut_kvp_open( pInstance, NULL, true );
+    UT_LOG_STEP("ut_kvp_openMemory( pInstance, NULL ) - Negative");
+    status = ut_kvp_openMemory( pInstance, NULL, -1);
     UT_ASSERT( status == UT_KVP_STATUS_INVALID_PARAM );
 
     /* data doesn't exist */
-    UT_LOG_STEP("ut_kvp_open( pInstance, %s - filename doesn't exist ) - Negative", KVP_VALID_TEST_NO_FILE);
+    UT_LOG_STEP("ut_kvp_openMemory( pInstance, %s - filename doesn't exist ) - Negative", KVP_VALID_TEST_NO_FILE);
     const char *kvp_str = "  ";
     char* kvp_buffer = strdup(kvp_str);
-    if (!kvp_buffer) {
-        UT_LOG_ERROR("strdup");
+    if (kvp_buffer)
+    {
+        status = ut_kvp_openMemory(pInstance, kvp_buffer, -1);
+        UT_ASSERT(status == UT_KVP_STATUS_PARSING_ERROR);
     }
-    status = ut_kvp_open( pInstance, kvp_buffer, true );
-    UT_ASSERT( status == UT_KVP_STATUS_PARSING_ERROR );
 
     /* Zero length file, that the library should reject because it can't parse it at all */
-    UT_LOG_STEP("ut_kvp_open( pInstance, %s - zero length file ) - Negative", KVP_VALID_TEST_ZERO_LENGTH_YAML_FILE);
-    status = ut_kvp_open( pInstance, read_file(KVP_VALID_TEST_ZERO_LENGTH_YAML_FILE), true );
-    UT_ASSERT( status == UT_KVP_STATUS_PARSING_ERROR );
+    UT_LOG_STEP("ut_kvp_openMemory( pInstance, %s - zero length file ) - Negative", KVP_VALID_TEST_ZERO_LENGTH_YAML_FILE);
+    if (read_file(KVP_VALID_TEST_ZERO_LENGTH_YAML_FILE, &pMemoryInstance) == 0)
+    {
+        status = ut_kvp_openMemory(pInstance, pMemoryInstance.buffer, pMemoryInstance.length);
```

Comment:
your not freeing your memory, when you open another file? Is that a feature of the function?

What happens if your trying to openMemory when the Memory is already open as part of the instance?

```
    if (read_file(KVP_VALID_TEST_JSON_FILE, &kvpMemory) == 0)
    assert( status == UT_KVP_STATUS_SUCCESS );
    {
        status = ut_kvp_openMemory(gpMainTestInstance, kvpMemory.buffer, kvpMemory.length);
        assert(status == UT_KVP_STATUS_SUCCESS);
    }
```

can therefore become

```c
status = ut_kvp_openMemory(gpMainTestInstance, gYamlData.buffer, gYamlData.length);
UT_ASSERT(status == UT_KVP_STATUS_SUCCESS);
```

Reply:
you should not be using assert() - in the tests you should be using `UT_ASSERT()`

Reply:
> your not freeing your memory, when you open another file? Is that a feature of the function?

Yes, the memory gets freed as part of the handle destroy(_fy_document_destroy()_)